### PR TITLE
Feature/setup commits validation and contributors readme

### PR DESCRIPTION
**Pull request by DanielaValero**
## What this PR do?
* Adds a contributor documentation, all the sections are taken from webpack and webpack.js.org
* Proposes using a subset of git-flow as branching model. Basically using prefixes for new features and bugfixes. More info in its own commit: https://github.com/webpack/webpack-cli/pull/3/commits/abb5f7a324eb955accf5d31428c7732ea2219fdf
* Adds an option to use yarn for the dependencies management
* Adds the editorcofig a js beautifier settings of webpack
* Adds a commit message validation
* Proposes a template for commit messages. Same as the one of angularJS, but without the scope. More info in its own commit: https://github.com/webpack/webpack-cli/pull/3/commits/eda75530805dc97adba5901d423e3ae92221fea3

## Where should the reviewer start?
* Checking the contributor documentation 
* Checking proposed branching model
* Checking the proposed commit messages template

### For the yarn setup
Just follow the installation instrucctions in the contributing.

### For the commit validation
Once you checked out the branch, do:
* `npm i`: Will install the new dependency
* `npm run install-commit-validator`: Will link to pre-commit hook the cli of the commit validator

## Where should the reviewer start?
Checking the contributor documentation

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+the api please create an issue first. This way we can ensure that your precious
+work is not in vain.
+
+## Issues
+
+Most of the time, when webpack does not work correctly, it might be a configuration issue.
+
+If you are still having difficulty after looking over your configuration carefully, please post
+a question to [StackOverflow with the webpack-cli tag](http://stackoverflow.com/tags/webpack-cli). Questions
+that include your webpack.config.js and relevant files, this way you help others to help you.
```
Comment (ev1stensberg) [inline CONTRIBUTING.md]: Use `webpack.config.js` as this is a filename

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+### How to increase chance of having a pull request accepted?
+
+* Write tests
+* Follow the existing coding style
+* Write a [good commit message](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
+
+## Documentation
+
+webpack is insanely feature rich and documentation is a time sink. We
```
Comment (ev1stensberg) [inline CONTRIBUTING.md]: Consider rephrasing the first sentence, might be somewhat unclear what you meant

**Comment by DanielaValero** [comment]
Checking the errors of travis

File: .gitignore [hunk 3]
```diff
@@ -7,4 +7,5 @@
 node_modules
 
 # npm-debug log
-npm-debug.log
\ No newline at end of file
+npm-debug.log
+yarn-error.log
```
Comment (ev1stensberg) [inline .gitignore]: Should be `#yarn-debug log` and the log should be placed under there or you change `#npm-debug log` to `npm and yarn-debug log`

File: CONTRIBUTING.md (hunk 1 above)
Comment (DanielaValero) [inline CONTRIBUTING.md]: Hi, I don't understand quite right what you mean here, would you be more specific about the change ?

Comment (ev1stensberg) [inline CONTRIBUTING.md]: Just replace webpack.config.js with `webpack.config.js` 👍

**Comment by ev1stensberg** [comment]
Thx! Nice work @DanielaValero !

**Pull request merged**